hello-world-master/source/main/components/KeyboardScreen/KeyboardScreen.bs

```diff
@@ -0,0 +1,98 @@
+import "../Services/NavigationService.bs"
+
+sub init()
+    m.background = m.top.findNode("background")
+    m.keyboard = m.top.findNode("keyboard")
+    m.searchResultsLabel = m.top.findNode("searchResultsLabel")
+    m.markupGrid = m.top.findNode("markupGrid")
+
+    m.keyboard.ObserveField("text", "onKeyboardTextChange")
+
+    m.navigationService = new NavigationService(m.top)
+end sub
+
+sub showContent(event as Object)
+    m.pokemonData = event.getData()
+    m.markupGrid.content = cloneContentNode(m.pokemonData)
+
+    m.background.color = "0x808080FF"
+    m.keyboard.setFocus(true)
+end sub
+
+sub onKeyboardTextChange(event as Object)
+    query = event.getData()
+    results = CreateObject("roSGNode","ContentNode")
+    childCount = m.pokemonData.getChildCount()
+    pokemonData = cloneContentNode(m.pokemonData)
+
+    if query = ""
+        m.markupGrid.content = pokemonData
+        m.searchResultsLabel.text = "Pokemons"
+
+        return
```

Comment:
A function should not have forced returns, only in special cases.
Let's use a big if instead of this one that forces the exit from function